### PR TITLE
Some example improvements

### PR DESCRIPTION
**Pull request by mitschabaude**
fixes https://github.com/o1-labs/o1js/issues/1228

start improving `/examples` folder by moving more files into their own folders, cleaning up a few examples, and adding some comments and readmes. Also fixes a type error in `ZkProgram.compile()`

I'm still not really settling on a folder structure I like, and there's still quite a bit of random junk left in `/examples`, but this is a start

### CODE REVIEW

File: src/examples/internals/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Examples: Internals
+
+This folder contains examples which highlight inner workings and less-documented behaviours of o1js.
+
+These examples might be useful for advanced users and contributors.
```
Comment (barriebyron) [inline src/examples/internals/README.md:5]:
```suggestion
This folder contains examples that highlight the inner workings and less-documented behaviors of o1js.

These examples might be useful for advanced users and contributors.

Run the examples with `npm run build`.
```

Comment (barriebyron) [reply src/examples/internals/README.md:5]: @mitschabaude It was only yesterday when I saw the description in https://github.com/o1-labs/o1js/issues/1222 that I knew we had runnable examples... we might not need the "Run the examples with..." here, but perhaps at a top-level?

Comment (mitschabaude) [reply src/examples/internals/README.md:5]: that's not correct :D but good idea to add an isntruction how to run examples! I'll add it to the top level examples folder

**Review by MartinMinkov — APPROVED** [review]

**Review by MartinMinkov — APPROVED** [review]

**Comment by barriebyron** [comment]
soooooo goood thank you!!

**Pull request merged**